src/Compilers/CSharp/Portable/Syntax/CSharpSyntaxTree.cs

```diff
@@ -302,7 +303,12 @@ private void BuildPreprocessorStateChangeMap()
         /// <summary>
         /// Creates a new syntax tree from a syntax node.
         /// </summary>
-        public static SyntaxTree Create(CSharpSyntaxNode root, CSharpParseOptions options = null, string path = "", Encoding encoding = null)
+        public static SyntaxTree Create(
+            CSharpSyntaxNode root,
+            CSharpParseOptions options = null,
+            string path = "",
+            Encoding encoding = null,
+            ImmutableDictionary<string, ReportDiagnostic> diagnosticOptions = null)
```

Comment:
this still feels super weird to me.  we have two sets of options for controlling things per tree.  Note: in many ways this will be a PITA for the IDE code.  For example, all our test helpers allow for providing a set of parse options.  Now they'll need this option as well just to flow this data along.  Now, if htat's really necessary for this design, that's fine.  But it's seems super odd to have two ways to specify options (to me at least).